hudi-flink-datasource/hudi-flink/src/test/java/org/apache/hudi/sink/meta/TestCkpMetadata.java

```diff
@@ -94,9 +107,17 @@ void testBootstrap() throws Exception {
         metadata1.getInstantCache(), is(Collections.singletonList("4")));
   }
 
-  private CkpMetadata getCkpMetadata(String uniqueId) {
-    String basePath = tempFile.getAbsolutePath();
-    FileSystem fs = FSUtils.getFs(basePath, HadoopConfigurations.getHadoopConf(new Configuration()));
-    return CkpMetadata.getInstance(fs, basePath, uniqueId);
+  protected CkpMetadata getCkpMetadata(String uniqueId) {
+    conf.set(FlinkOptions.WRITE_CLIENT_ID, uniqueId);
+    return CkpMetadataFactory.getCkpMetadata(writeClient.getHoodieTable().getMetaClient(), writeClient.getConfig(), conf);
+  }
```

Comment:
Create a hoodie table each time is costly.

Reply:
Changed to use metaClient created in `setup` method instead

Reply:
Maybe just pass around the write config should be enough,

Reply:
Using a metaclient and a writeConfig to initialize the CkpMetadata is suitable for Coordinator and write functions.  Maybe we don't need to add a new constructor for `CkpMetadata` and use the constructor that write functions used when testing.